features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/APIClient.js

```diff
@@ -51,11 +53,17 @@ class SingleClient {
                         requestInterceptor: this._getRequestInterceptor(),
                         responseInterceptor: this._getResponseInterceptor()
                     });
-                return new Swagger(argsv);
+                return new SwaggerClient(argsv);
             }
         );
-        this._client.catch(AuthManager.unauthorizedErrorHandler);
-        SingleClient._instance = this;
+        promisedSwaggerClient.catch(AuthManager.unauthorizedErrorHandler);
+
+        this._client = promisedSwaggerClient.then(swaggerClient => {
+            swaggerClient.http.withCredentials = true;
```

Comment:
I think we can move setting `swaggerClient.http.withCredentials = true;` inside the previous then handler just after line #56 ?

Reply:
In swagger-js docs it is.
```js
var swaggerClient = new SwaggerClient(specUrl)
      .then(function (swaggerClient) {                                            
          swaggerClient.http.withCredentials = true; // this activates CORS, if necessary
          return swaggerClient.apis.pet.addPet({id: 1, name: "bobby"}); // chaining promises
      });
```
I think it should be with the response `swaggerClient`, after initilizing `SwaggerClient`.

Reply:
Fixed with the commit https://github.com/RenukaFernando/carbon-apimgt/commit/f23f437254d2faae448a18be410a07137fb7689b